runtime/src/main/java/org/corfudb/util/CFUtils.java

```diff
@@ -145,6 +150,37 @@ public static void runAfter(Duration duration, Runnable toRun) {
         return CompletableFuture.allOf(futuresArr);
     }
 
+    /**
+     * Takes a list of the completable futures and returns the CompletableFuture of a list.
+     *
+     * @param futures A list of completable futures, perhaps a result of a map function.
+     * @param <T>     A return type of the future.
+     * @return A completable future, which completes with a list of the results.
+     */
+    public static <T> CompletableFuture<List<T>> sequence(List<CompletableFuture<T>> futures) {
+        return CompletableFuture
+                .allOf(futures.toArray(new CompletableFuture<?>[futures.size()]))
```

Comment:
Would be great if you replace by non-blocking version
```
public static <T> CompletableFuture<List<T>> sequence(List<CompletableFuture<T>> futures) {

        return allOf(futures).thenCompose(empty -> {
                    CompletableFuture<List<T>> aggregated = CompletableFuture
                            .completedFuture(new ArrayList<>());

                    for (CompletableFuture<T> future : futures) {
                        aggregated = aggregated.thenCombine(future, (List<T> list, T value) -> {
                            list.add(value);
                            return list;
                        });
                    }

                    return aggregated;
                }
        );
    }
```

testing:
```
public static void main(String[] args) {
        List<CompletableFuture<Integer>> list = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            final int currVal = i;
            list.add(CompletableFuture.supplyAsync(() -> {
                Sleep.sleepUninterruptibly(Duration.ofMillis(100));
                return currVal;
            }));
        }

        CompletableFuture<List<Integer>> result = CFUtils.sequence(list);
        System.out.println(result.join());
    }
```
prints:
```
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Reply:
K